tests/test_plugins/autograd/test_functions.py

```diff
@@ -300,3 +302,38 @@ def test_threshold_exceptions(array, vmin, vmax, level, expected_message):
     """Test threshold function for expected exceptions."""
     with pytest.raises(ValueError, match=expected_message):
         threshold(array, vmin, vmax, level)
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3, 4])
+@pytest.mark.parametrize("method", ["linear", "nearest"])
+class TestInterpn:
```

Comment:
What do these test classes do? want to learn 😆

Reply:
It's really just a way to group related tests so that the tests contained in this class have access to some common methods (like the array construction here) and possibly fixtures that are only defined within the scope of one test set. So here I'm grouping the tests for interp that check equivalence to scipy and those that check grads because they can both reuse the same array setup.
The whole class is also parametrized such that all methods contained within that test class will be checked according to that common parametrization. In this case that's dimensionality and interpolation method.
The autograd test is then further sub-parametrized to check gradient order and mode.

Reply:
ah, that's interesting. I could probably use that in the main autograd tests, maybe I'll give it a try.